### PR TITLE
CORE-6057: Add down/error component name to SubscriptionDominoTileBase log

### PR DESCRIPTION
**Pull request by yift-r3**
This PR follow the status change of each child of the `SubscriptionDominoTileBase`, so we can log which child went down.

It also remove the resource after closing it to prevent reclosing.

### CODE REVIEW

**Review by fowlerrr — DISMISSED** [review]
LifecycleProcessor changes look okay

File: components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt [hunk 1]
```diff
@@ -128,46 +130,51 @@ abstract class SubscriptionDominoTileBase(
...
+                    logger.info("All dependencies are started now, starting subscription.")
+                    createAndStartSubscription()
+                } else {
+                    logger.info("The status of $name had started. Waiting for $notReady.")
+                }
+            }
+            LifecycleStatus.DOWN, LifecycleStatus.ERROR -> {
+                logger.info("The status of $name changed from $oldStatus to $status, stopping subscription.")
+                subscriptionRegistration.getAndSet(null)?.close()
+                updateState(StoppedDueToChildStopped)
```
Comment (williamvigorr3) [inline components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt]: We no longer set `StoppedDueToError`, if `LifecycleStatus.ERROR`. Looks like this is the correct behavour, going off what complex domino tile does.

Comment (williamvigorr3) [reply components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt]: Just checking you made this change on purpose?

Comment (yift-r3) [reply components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt]: Sorry, missed that comment...

**Review by williamvigorr3 — APPROVED** [review]
LGTM

File: components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt [hunk 2]
```diff
@@ -128,46 +130,55 @@ abstract class SubscriptionDominoTileBase(
...
+                    logger.info("All dependencies are started now, starting subscription.")
+                    createAndStartSubscription()
+                } else {
+                    logger.info("The status of $name had started. Waiting for $notReady.")
+                }
+            }
+            LifecycleStatus.DOWN, LifecycleStatus.ERROR -> {
+                logger.info("The status of $name changed from $oldStatus to $status, stopping subscription.")
+                subscriptionRegistration.getAndSet(null)?.close()
+                if (status == LifecycleStatus.ERROR) {
```
Comment (williamvigorr3) [inline components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt:174]: Should this be reversed?

**Review by williamvigorr3 — CHANGES_REQUESTED** [review]

File: components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt (hunk 2 above)
Comment (yift-r3) [reply components/domino-logic/src/main/kotlin/net/corda/lifecycle/domino/logic/util/SubscriptionDominoTileBase.kt:174]: Yes, it should

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Pull request merged**